src/source/rtl_text_plugin.js

```diff
@@ -55,23 +78,73 @@ export const setRTLTextPlugin = function(url: string, callback: ErrorCallback) {
             }
         } else {
             // Called once for each worker
-            pluginStatus = status.loaded;
+            if (!lazy) {
+                pluginStatus = status.loaded;
+            }
         }
     };
-    evented.fire(new Event('pluginAvailable', {pluginURL, completionCallback: _completionCallback}));
+
+    if (lazy) {
+        // Inform the worker-threads that we intend to load the plugin lazily later,
+        // This is so the workers can skip RTL text parsing.
+        evented.fire(new Event('pluginAvailable', {
+            pluginURL: null,
+            lazy,
+            completionCallback: _completionCallback
+        }));
+    } else {
+        downloadRTLTextPlugin();
+    }
+};
+
+export const downloadRTLTextPlugin = function() {
+    if (pluginStatus !== status.available) {
+        throw new Error('rtl-text-plugin cannot be downloaded unless a pluginURL is specified');
+    }
+    pluginStatus = status.loading;
+
+    if (pluginURL) {
+        getArrayBuffer({url: pluginURL}, (error, data) => {
+            if (error || !data) {
+                throw error;
+            } else {
+                const rtlBlob = new window.Blob([data], {type: 'application/javascript'});
+                const URL = window.URL || window.webkitURL;
```

Comment:
why do we need `webkitURL` here? i can't even really find much information on it but it seems to be deprecated

Reply:
This may have been added in the past when Safari/Webkit didn't support `window.URL`. But that doesn't seem to be an issue any longer based on [caniuse](https://caniuse.com/#search=createObjectURL)